tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_opcodes_in_legacy.py

```diff
@@ -0,0 +1,86 @@
+"""
+Tests all EOF-only opcodes in legacy contracts and expects failure.
+"""
+
+import pytest
+
+from ethereum_test_base_types import Alloc, Account
+from ethereum_test_specs import StateTestFiller
+from ethereum_test_tools import Bytecode, EOFException, EOFTestFiller
+from ethereum_test_tools import Opcodes as Op
+from ethereum_test_tools.eof.v1 import AutoSection, Container, Section
+from ethereum_test_types import Environment, Transaction
+from ethereum_test_vm import Opcodes
+
+from .. import EOF_FORK_NAME
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-7692.md"
+REFERENCE_SPEC_VERSION = "f0e7661ee0d16e612e0931ec88b4c9f208e9d513"
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+slot_code_executed = b"EXEC"
+value_code_executed = b"exec"
+value_non_execution_canary = b"brid"
+
+
+@pytest.mark.parametrize(
+    "code",
+    [
+        pytest.param(Op.PUSH0 + Op.DUPN[0], id="DUPN"),
+        pytest.param(Op.PUSH0 + Op.SWAPN[0], id="SWAPN"),
+        pytest.param(Op.PUSH0 + Op.PUSH0 + Op.PUSH0 + Op.EXCHANGE[2, 3], id="EXCHANGE"),
+        pytest.param(Op.RJUMP[0], id="RJUMP"),
+        pytest.param(Op.PUSH0 + Op.RJUMPI[0], id="RJUMPI"),
+        pytest.param(Op.PUSH0 + Op.RJUMPV[0, 0], id="RJUMPI"),
+        pytest.param(Op.CALLF[1], id="CALLF"),
+        pytest.param(Op.RETF, id="RETF"),
+        pytest.param(Op.JUMPF, id="JUMPF"),
+        pytest.param(Op.PUSH0 + Op.PUSH0 + Op.PUSH0 + Op.PUSH1(2) + Op.EXTCALL, id="EXTCALL"),
+        pytest.param(
+            Op.PUSH0 + Op.PUSH0 + Op.PUSH0 + Op.PUSH1(2) + Op.EXTDELEGATECALL, id="EXTDELEGATECALL"
+        ),
+        pytest.param(
+            Op.PUSH0 + Op.PUSH0 + Op.PUSH0 + Op.PUSH1(2) + Op.EXTSTATICCALL, id="EXTSTATICCALL"
+        ),
+        pytest.param(Op.DATALOAD(0), id="DATALOAD"),
+        pytest.param(Op.DATALOADN[0], id="DATALOADN"),
+        pytest.param(Op.DATASIZE, id="DATASIZE"),
+        pytest.param(Op.DATACOPY(0, 0, 32), id="DATACOPY"),
+        pytest.param(Op.EOFCREATE[0](0, 0, 0, 0), id="EOFCREATE"),
+        pytest.param(Op.RETURNCONTRACT[0], id="RETURNCONTRACT"),
+    ],
+)
+def test_opcodes_in_legacy(state_test: StateTestFiller, pre: Alloc, code: Opcodes):
```

Comment:
I think we should repeat this test in a couple more contexts:
- Contract creating TX initcode
- `CREATE` opcode initcode
- `CREATE2` opcode initcode
- EXTCALL to legacy contract coming from EOF contract

It might be easier to define the list of `pytest.param` as a global list for this file and then use `@pytest.mark.parametrize` to repeat the bytecodes with the invalid opcodes in different test functions for each context.

Reply:
Done